src/Processors/Executors/PipelineExecutor.h

```diff
@@ -67,8 +67,8 @@ class PipelineExecutor
     // Concurrency control related
     ConcurrencyControl::AllocationPtr slots;
     ConcurrencyControl::SlotPtr single_thread_slot; // slot for single-thread mode to work using executeStep()
-    std::mutex threads_mutex;
-    std::vector<ThreadFromGlobalPool> threads;
+    ThreadPool pool;
```

Comment:
Doesn't it change behaviour somehow? Was there any reason to use global pool here? @KochetovNicolai

Reply:
I've decided to do this separately. Rebased without this patch and to fix conflicts.

Reply:
#48146